### PR TITLE
Initial checkin

### PR DESCRIPTION
**Pull request by bwalsh**
This PR:

* creates a command line UX for `gen3_util`
* includes tests and source code conventions
* distributes to pypi

```
$gen3_util --help

Usage: gen3_util [OPTIONS] COMMAND [ARGS]...

  Gen3 Management Utilities

Options:
  --config TEXT              Path to config file. GEN3_UTIL_CONFIG
  --format [yaml|json|text]  Result format. GEN3_UTIL_FORMAT
  --help                     Show this message and exit.

Commands:
  projects  Manage Gen3 projects.
  meta      Manage meta data.
  files     Manage file buckets.
  config    Configure this utility.

```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,78 @@
...
+
+## Distribution
+
+- PyPi
+
+```
+# update pypi
+
+export TWINE_USERNAME=  #  the username to use for authentication to the repository.
+export TWINE_PASSWORD=  # the password to use for authentication to the repository.
```
Comment (lbeckman314) [inline README.md:73]: Looks great! Should developers setup individual [Twine](https://twine.readthedocs.io/en/stable/) credentials or does this refer to the [PyPi](https://pypi.org/) accounts?

**Comment by bwalsh** [comment]
```
$ gen3_util projects

Usage: gen3_util projects [OPTIONS] COMMAND [ARGS]...

  Manage Gen3 projects.

Options:
  --help  Show this message and exit.

Commands:
  ls     List all projects.
  touch  Create a project
  rm     Remove project.
$ gen3_util meta

Usage: gen3_util meta [OPTIONS] COMMAND [ARGS]...

  Manage meta data.

Options:
  --help  Show this message and exit.

Commands:
  ls        List meta in a project.
  validate  Copy meta to/from the project bucket.
  cp        Copy meta to/from the project bucket.
  rm        Remove meta from a project.

$ gen3_util files

Usage: gen3_util files [OPTIONS] COMMAND [ARGS]...

  Manage file buckets.

Options:
  --help  Show this message and exit.

Commands:
  ls  List files in a project.
  cp  Copy files to/from the project bucket.
  rm  Remove files from a project.

$ gen3_util config

Usage: gen3_util config [OPTIONS] COMMAND [ARGS]...

  Configure this utility.

Options:
  --help  Show this message and exit.

Commands:
  ls  Show defaults.
$ 
```

File: gen3_util/files/downloader.py
Comment (lbeckman314) [inline gen3_util/files/downloader.py:1]: The command structure is really well organized. Will the `downloader` module be imported into `gen3_util/files/cli.py` like `uploader` above, or will it be used differently?

File: gen3_util/meta/cli.py
Comment (lbeckman314) [inline gen3_util/meta/cli.py:1]: Looks great! Will the `meta` and `files` packages share the same logic for upload/download/list/remove or will they follow different patterns?

**Review by lbeckman314 — APPROVED** [review]
This looks awesome! Couple questions regarding the following but nothing getting in the way of merging the PR:
- Twine credentials
- `downloader` vs `uploader` import
- `meta` vs `files` logic

File: README.md (hunk 1 above)
Comment (lbeckman314) [reply README.md:73]: Ah, no Twine accounts just credentials for the`PyPI` repository if I understand correctly. Cool!

Comment (bwalsh) [reply README.md:73]: Correct

File: gen3_util/meta/cli.py
Comment (bwalsh) [reply gen3_util/meta/cli.py:1]: Meta and files will be different, exact details to be worked out

File: gen3_util/files/downloader.py
Comment (bwalsh) [reply gen3_util/files/downloader.py:1]: It will, I'll change before merging

Comment (bwalsh) [reply gen3_util/files/downloader.py:1]:
```gen3_util files cp --help
Usage: gen3_util files cp [OPTIONS] FROM TO

  Copy files to/from the project bucket.

  Args:     FROM: Source url or path     TO: Destination url or path

Options:
  --help  Show this message and exit.
```

**Pull request merged**